src/main/java/songbox/house/service/impl/GoogleDriveServiceImpl.java

```diff
@@ -23,28 +28,99 @@
 @AllArgsConstructor
 @FieldDefaults(makeFinal = true, level = PRIVATE)
 public class GoogleDriveServiceImpl implements GoogleDriveService {
-
+    private static final String FOLDER_NAME = "SongboxHouse";
+    private static final String DEFAULT_COLLECTION_FOLDER = "DefaultCollection";
     private static final String MIME_TYPE = "audio/mpeg";
+    public static final String GOOGLE_DRIVE_MIMETYPE_FOLDER = "application/vnd.google-apps.folder";
 
     GoogleAuthenticationService authenticationService;
+    UserPropertyService userPropertyService;
 
-    @Override
-    public void upload(Track track) {
-        final Drive drive = authenticationService.getDrive();
+    private File getFolderOrCreate(Drive drive, String folderName, File parentFolder) throws IOException {
```

Comment:
the new code in this class is not clear enough for me, I hope you tested it and it works as it should